src/GraphQL.hs

```diff
@@ -54,6 +69,61 @@ instance GraphQLError QueryError where
     "Validation errors:\n" <> mconcat ["  " <> formatError e <> "\n" | e <- NonEmpty.toList es]
   formatError (ExecutionError e) =
     "Execution error: " <> show e
+  formatError (NonObjectResult v) =
+    "Query returned a value that is not an object: " <> show v
+
+-- | Execute a GraphQL query.
+executeQuery
+  :: forall api m. (HasGraph m api, Applicative m)
```

Comment:
We discussed adding an `Object n i f ~ api` constraint to make sure `executeQuery` can run only on Objects. Up to you.